### PR TITLE
Add documentation for Using Components

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR adds some basic documentation for how to use the different categories of components provided by the package and their APIs.

Run `make dev` and visit http://localhost:4001/using-components

image

Depends on #505

### CODE REVIEW

File: src/pattern-library/components/patterns/UsingComponents.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * @prop {import('preact').ComponentChildren} [children]
+ * @prop {string|string[]} [classes] - Optional extra CSS classes to append to the
+ *   component's default classes
+ * @prop {never} [className] - Use variants, props, unstyled component (when
+ *   available) or classes instead
+ * @prop {import('preact').Ref<HTMLElement>} [elementRef] - Ref for component's
+ *   outermost element.
+ */
+`}
+        />
```
Comment (lyzadanger) [inline src/pattern-library/components/patterns/UsingComponents.js:75]: There are about fifteen hundred more elegant ways to pull this off but...it works for the moment.

I had initially written out prose descriptions of the common props but found that nothing was as insightful as the typedefs themselves.

File: src/pattern-library/components/patterns/UsingComponents.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import Library from '../Library';
+
+import Next from '../LibraryNext';
+
+export default function UsingComponentsPage() {
+  return (
+    <Library.Page
+      title="Using components"
+      intro={
+        <>
```
Comment (robertknight) [inline src/pattern-library/components/patterns/UsingComponents.js:10]: The fragment is currently redundant here.

File: src/pattern-library/components/patterns/UsingComponents.js (hunk 1 above)
Comment (robertknight) [reply src/pattern-library/components/patterns/UsingComponents.js:75]: One way we could address this is with convenient links from the component library web pages to the relevant part of the source code, either on GitHub (easiest solution) or rendered by the dev server itself (more work, especially if we wanted to serve this as a static site).

**Review by robertknight — APPROVED** [review]
LGTM with just a couple of minor comments.

**Comment by lyzadanger** [comment]
@robertknight : I was a little trigger-happy with merge on this PR, apologies; I would like to combine your minor feedback with a documentation-related request in another PR as a small follow-up.

**Pull request merged**